torch/nn/functional.py

```diff
@@ -3223,8 +3223,9 @@ def multi_head_attention_forward(query,                           # type: Tensor
             be ignored by the attention. This is an binary mask. When the value is True,
             the corresponding value on the attention layer will be filled with -inf.
         need_weights: output attn_output_weights.
-        attn_mask: mask that prevents attention to certain positions. This is an additive mask
-            (i.e. the values will be added to the attention layer).
+        attn_mask: 2D or 3D mask that prevents attention to certain positions. This is an additive mask
+            (i.e. the values will be added to the attention layer). A 2D mask will be broadcasted for all
+            the batches while a 3D mask allows a different mask on each batch.
```

Comment:
Nit: I think what you mean here is "A 2D mask will be broadcasted for all entries of the given batch while a 3D mask allows a different mask on each entry".

Reply:
It's actually batch-wise. Each batch has an attention mask in the 3D case.

Reply:
So there's a batch of batches?